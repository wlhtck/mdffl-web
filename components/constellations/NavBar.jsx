import React from 'react';
import { arrayOf, exact, string } from 'prop-types';
import { map } from 'lodash/fp';
import { primary } from '../nebulas/colors';
import { Row } from '../nebulas/styled-shoelaces';
import Image from '../nebulas/Image';
import NavLink from '../stars/NavLink';
import Button from '../stars/Button';
import withStyles from '../util/withStyles';

const navSpacing = { margin: '0 2vw' };
const NavItem = withStyles(navSpacing, NavLink);
const NavButton = withStyles(navSpacing, Button);
const NavLogo = withStyles({ margin: '0 auto 0 0' }, ({ className }) => (
  <a href="/" className={className}>
    <Image src="/mdffl-logo.png" alt="MDFFL Logo" />
  </a>
));

const NavBar = withStyles({
  padding: '25px',
  backgroundColor: primary,
}, ({ links, className }) => (
  <Row alignItems="center" className={className} gutter="0">
    <NavLogo />
    {map((link) => <NavItem {...link} />, links)}
    <NavButton
      href="https://www.playyon.com/metro-detroit-flag-football-league/registrations/"
      type="secondary"
      external
    >
      REGISTER
    </NavButton>
  </Row>
));

NavBar.propTypes = {
  links: arrayOf(exact({
    key: string.isRequired,
    href: string.isRequired,
    children: string.isRequired,
  })).isRequired,
  className: string,
};

NavBar.defaultProps = {
  className: '',
};

export default NavBar;
