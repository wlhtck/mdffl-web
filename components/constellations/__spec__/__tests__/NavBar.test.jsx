import React from 'react';
import 'jest-styled-components';
import { render } from 'enzyme';
import NavBar from '../../NavBar';

describe('conestellations/NavBar', () => {
  it('renders a <NavBar /> with some links', () => {
    expect(render(
      <NavBar
        links={[
          { href: '/', children: 'About', key: 'about' },
          { href: '/', children: 'Home', key: 'home' },
        ]}
      />,
    )).toMatchSnapshot();
  });
});
