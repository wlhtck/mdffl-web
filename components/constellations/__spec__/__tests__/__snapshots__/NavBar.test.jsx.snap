// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conestellations/NavBar renders a <NavBar /> with some links 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 *:only-child {
  margin: 0 auto;
}

.c1 *:last-child {
  margin-bottom: 0;
}

.c3 {
  color: #FFFFFF;
  font-family: 'Raleway',sans-serif;
  font-size: 16px;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  font-family: 'Raleway',sans-serif;
  font-size: 16px;
  font-weight: 800;
  background-color: #FFFFFF;
  color: #1C104E;
  display: inline-block;
  padding: 16px;
  border: none;
  line-height: 1;
  border-radius: 4px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  padding: 25px;
  background-color: #1C104E;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c2 {
  margin: 0 auto 0 0;
}

.c0 > * {
  margin: 0 32px;
}

<NavBar
  links={
    Array [
      Object {
        "key": "about",
        "text": "About",
        "url": "/",
      },
      Object {
        "key": "home",
        "text": "Home",
        "url": "/",
      },
    ]
  }
>
  <NavBar
    className="c0"
    links={
      Array [
        Object {
          "key": "about",
          "text": "About",
          "url": "/",
        },
        Object {
          "key": "home",
          "text": "Home",
          "url": "/",
        },
      ]
    }
  >
    <Row
      className="c0"
      padding={null}
      reverse={false}
    >
      <div
        className="c0 c1"
      >
        <Image__StyledImage
          alt="MDFFL Logo"
          fullWidth={false}
          src="/mdffl-logo.png"
        >
          <img
            alt="MDFFL Logo"
            className="c2 "
            src="/mdffl-logo.png"
          />
        </Image__StyledImage>
        <NavLink
          key="about"
          text="About"
          url="/"
        >
          <NavLink
            className="c3"
            text="About"
            url="/"
          >
            <a
              className="c3"
              href="/"
            >
              About
            </a>
          </NavLink>
        </NavLink>
        <NavLink
          key="home"
          text="Home"
          url="/"
        >
          <NavLink
            className="c3"
            text="Home"
            url="/"
          >
            <a
              className="c3"
              href="/"
            >
              Home
            </a>
          </NavLink>
        </NavLink>
        <Button__StyledButton
          external={true}
          text="Register"
          type="secondary"
          url="https://google.com"
        >
          <Button
            className="c4"
            external={true}
            onClick={null}
            text="Register"
            type="secondary"
            url="https://google.com"
          >
            <a
              className="c4"
              href="https://google.com"
              rel="noreferrer noopener"
              target="_blank"
            >
              Register
            </a>
          </Button>
        </Button__StyledButton>
      </div>
    </Row>
  </NavBar>
</NavBar>
`;
